lib/rules/selector-not-notation/index.mjs

```diff
@@ -82,43 +85,38 @@ const rule = (primary, _, context) => {
 			selectorRoot.walkPseudos((pseudo) => {
 				if (!isNot(pseudo)) return;
 
+				const range = getRange(pseudo, ruleNode);
+				let fix;
+
 				if (primary === 'complex') {
 					const prev = pseudo.prev();
 					const hasConsecutiveNot = prev && isNot(prev);
 
 					if (!hasConsecutiveNot) return;
 
-					if (context.fix) return fixComplex(prev);
+					fix = fixComplex.bind({ ruleNode, selectorRoot }, prev);
 				} else {
 					const selectors = pseudo.nodes;
 
 					if (isSimple(selectors)) return;
 
-					const mustFix =
-						context.fix &&
-						selectors.length > 1 &&
-						selectors[1] &&
-						(selectors[1].nodes.length === 0 || selectors.every(({ nodes }) => nodes.length === 1));
+					const second = selectors.length > 1 && selectors[1];
+					const isFixable =
+						second && (!second.nodes.length || selectors.every(({ nodes }) => nodes.length === 1));
 
-					if (mustFix) return fixSimple(pseudo);
+					if (isFixable) fix = fixSimple.bind({ ruleNode, selectorRoot }, pseudo);
```

Comment:
[suggest] Same here. https://github.com/stylelint/stylelint/pull/7755/files#r1637409423

```js
if (isFixable) {
	fix = () => {
		const offset = fixSimple(pseudo);
		ruleNode.selector = selectorRoot.toString();
		return getNewRange(pseudo, ruleNode, offset);
	};
}
```

Reply:
I am not convinced that's better.
What about not using `this`?
i.e. adding 2 new parameters instead
e.g. `fixSimple.bind(null, pseudo, ruleNode, selectorRoot);`

Reply:
Instead of `bind`, can we just do this? ⬇️ 

```js
fix = () => fixSimple(pseudo, ruleNode, selectorRoot)
```